project/henry0928/README.md

```diff
@@ -0,0 +1,113 @@
+The `Beach Line` consists of the parabola formed by all the input points included in the calculation.
+
+### Event:
+
+The two major events that have impact on the diagram are the following:
+
+1. `Site Event`: Occurs when the Sweep Line passes through any input point, and the passed input point will be included in the calculation and change the Beach Line to form the sides of the Voronoi diagram.
+2. `Circle Event`: Occurs when any parabolic segment on the Beach Line disappears. At this time, there will be a special point that reaches 3 different site points and the Sweep Line is equidistant. This special point forms the vertex of the Voronoi Cell
+
+### Pseudo Code:
+```
+create the event queue with site events for each input points
+ 
+while the event queue is not empty
+ 
+    if the first event in the event queue is a site event
+        add the new parabola to the beach line
+ 
+    if the first event in the event queue is a circle event
+        add the new vertex and squeeze the parabola
+ 
+finish any remaining edges
+```
+# Prospective Users
+
+Anyone who wants to generate Voronoi Diagram and Delaunay Triangulation.
+
+# System Architecture
+
+This is a software system of a hybrid system of C++11 (modern C++) and Python.
+- Use `Python` to read input (target points, the boundary of the planar graph) and show results.
+- Use `C++` to implement "Fortune's Algorithm" and then generate Voronoi Diagram.
+- Use `Pybind11` to wrap C++ functions for Python.
+
+# API Description
+
+ The `result_diagram` here does not refer to the complete graph, but refers to the information containing all the edges and site points in the `result_diagram`, and draws a complete graph by using these information.
+
+`Python` API: <br>
+``` py 
+def Create_diagram(site_list, bb_point1, bb_point2) # Input the site's list and bounding box points to get the Voronoi Diagram as the result_diagram 
+def Draw_diagram(result_diagram) # Draw the result diagram
+```
+
+`Voronoi()` will get the edge information of Voronoi Diagram, user can use this function to get the edge information rathen than the complete graph. 
+
+`C++` API: <br>
+``` cpp
+Edge Voronoi(vector<Point> site_list, Point bb_point1, Point bb_point2 )
+```
+
+( If it is found that the API needs to be added during the development process, I will modify this part again. )
+
+# Engineering Infrastructure
+
+- Build: `make`
+- Version control: `git`
+- Testing infrastructure: `pytest`
+- Documentation: `README.md`
+
+# Schedule
+
+- Week 1 (4/10)
+    - Study the related knowledge, algorithm and finish proposal
+- Week 2 (4/17)
+    - Implement `Point`, `Edge`, `BeachLine` and `EventQueue` datastructures (C++)
+- Week 3 (4/24)
+    - Add functionality to `EventQueue` (C++)
+- Week 4 (5/01)
+    - Add functionality to `BeachLine` (C++)
+- Week 5 (5/08)
+    - Testing and debugging (C++)
```

Comment:
Although it is not too late, it will be more productive to move testing earlier in the plan.

Reply:
OK. I will think about it and modify it.